cluster-autoscaler/main.go

```diff
@@ -252,6 +255,16 @@ func createAutoscalingOptions() config.AutoscalingOptions {
 	if *maxDrainParallelismFlag > 1 && !*parallelDrain {
 		klog.Fatalf("Invalid configuration, could not use --max-drain-parallelism > 1 if --parallel-drain is false")
 	}
+
+	var parsedSchedConfig *scheduler_config.KubeSchedulerConfiguration
+	// if scheduler config flag was set by the user
+	if pflag.CommandLine.Changed(config.SchedulerConfigFileFlag) {
```

Comment:
This is so that if the `--scheduler-config-file` flag is not used, we don't try to look for the scheduler config in the default path set by the flag i.e., `""`. We want to use the default scheduler config. Setting the scheduler config to `nil` [ensures that the default scheduler config is used](https://github.com/kubernetes/autoscaler/pull/5708/files#diff-5818325464670a5cd54280fe253b0841a163da5de2764491a3065805785ffb1bR51-R56).

Reply:
I think there are two lines of thought here:
1. User has set `--scheduler-config-file=""` (note that this is a file path) which means they have made a mistake. Hence, fail. 
2. User has set `--scheduler-config-file=""` which means they don't want to set a scheduler config by themselves. Fall back to using a default one. 

Falling back to default in case of 2 doesn't seem as valuable to me because if the user wants to use the default config, they can just skip the flag altogether because the flag specifies path to a config file (and not scheduler config itself) and if they have set it to `""`, I think it's more likely that they have made a mistake (which can happen if say this value comes from a helm chart file and it's set to empty there by mistake) because they can just skip specifying the flag if they want to use the default config. 

A counter-argument could be, if the user sets only one or two fields in the custom scheduler config we anyway fill rest of the required fields by default (in this PR). It can be argued that 2 is inline with this behavior. 

I think failing here makes more sense than using default config but I am open to change it if you think there's a strong reason for not doing so.

Reply:
I was thinking about someone templating the manifest and using empty string to mean "default", but I guess such templating can also just skip the flag in such cases, so maybe it is fine to just leave as is.